src/main/scala/org/broadinstitute/dsde/firecloud/webservice/LibraryApiService.scala

```diff
@@ -49,15 +49,27 @@ trait LibraryApiService extends HttpService with FireCloudRequestBuilding
               }
             }
           } ~
-            path(Segment / Segment / "metadata") { (namespace, name) =>
-              put {
-                entity(as[String]) { rawAttrsString => requestContext =>
-                  perRequest(requestContext,
-                    LibraryService.props(libraryServiceConstructor, userInfo),
-                    LibraryService.UpdateAttributes(namespace, name, rawAttrsString.toString.parseJson))
-                }
+          path(Segment / Segment / "metadata") { (namespace, name) =>
+            put {
+              entity(as[String]) { rawAttrsString => requestContext =>
+                perRequest(requestContext,
+                  LibraryService.props(libraryServiceConstructor, userInfo),
+                  LibraryService.UpdateAttributes(namespace, name, rawAttrsString.toString.parseJson))
               }
             }
+          } ~
+          path(Segment / Segment / "published" ) { (namespace, name) =>
+            post { requestContext =>
+              perRequest(requestContext,
+                LibraryService.props(libraryServiceConstructor, userInfo),
+                LibraryService.SetPublishAttribute(namespace, name, true))
+            }
+            delete { requestContext =>
+              perRequest(requestContext,
+                LibraryService.props(libraryServiceConstructor, userInfo),
+                LibraryService.SetPublishAttribute(namespace, name, false))
+            }
+          }
```

Comment:
these three endpoints could be consolidated a bit, moving the namespace/name to the outside. Something like:
```
path(Segment / Segment) { (namespace, name) =>
  path("metadata") {
    put {
      ...
    }
  } ~
  path("published") {
    post {
      ...
    } ~
    delete {
      ...
    }
  }
}


```